core/src/main/java/org/elasticsearch/common/lucene/store/ByteArrayIndexInput.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.common.lucene.store;
+
+import org.apache.lucene.store.IndexInput;
+
+import java.io.EOFException;
+import java.io.IOException;
+
+/**
+ * Wraps array of bytes into IndexInput
+ */
+public class ByteArrayIndexInput extends IndexInput {
+    private final byte[] bytes;
+
+    private int pos;
+
+    private int offset;
+
+    private int length;
```

Comment:
The IndexInput API is using `long`. But here I am switching to `int` because I am dealing with a byte array and array indices have type `int` and it's not possible to create an array with size specified by `long` value. Something like this `new byte[3L]` will not even compile. In practice, the maximum array size is even smaller. Depending on java VM implementation but it can be between `Integer.MAX_VALUE-2` and `Integer.MAX_VALUE-5`. So, I think we should be fine in terms of both theoretical and practical limits but after your comment I have updated all comparisons in ByteArrayIndexInput to use `long` type to makes sure we don't miss integer overflow. Thanks!